rasa/core/policies/rule_policy.py

```diff
@@ -290,6 +448,8 @@ def _does_rule_match_state(rule_state: State, conversation_state: State) -> bool
                     # it should be None or non existent in the state
                     value == SHOULD_NOT_BE_SET
                     and conversation_sub_state.get(key)
+                    # this is needed to test on rule snippets
```

Comment:
Why is this only needed during testing?

Reply:
because during prediction `SHOULD_NOT_BE_SET` doesn't exist

Reply:
then let's phrase the comment as something like `during training `SHOULD_NOT_BE_SET` is provided and hence we also have to check for the value of the slot state`